src/main/java/org/runejs/client/Class35.java

```diff
@@ -11,118 +11,119 @@ public class Class35 implements Interface3 {
     public static IndexedImage aClass40_Sub5_Sub14_Sub2_1744;
     public static byte[][][] tile_overlay_rotation;
 
-    public LinkedList aLinkedList_1727 = new LinkedList();
-    public EntryTable[] aClass40_Sub10Array1740;
-    public int anInt1748;
-    public CacheArchive aCacheArchive_1749;
-    public double aDouble1750 = 1.0;
+    public LinkedList textureCache = new LinkedList();
+    public Texture[] textures;
+    public int textureSize;
+    public CacheArchive gameImageCacheArchive;
+    public double brightness = 1.0;
     public int anInt1752;
     public int anInt1753;
 
-    public Class35(CacheArchive arg0, CacheArchive arg1, int arg2, double arg3, int arg4) {
-        anInt1748 = 128;
+    public Class35(CacheArchive gameTextureCacheArchive, CacheArchive gameImageCacheArchive, int arg2, double arg3, int arg4) {
+        textureSize = 128;
         anInt1753 = 0;
         anInt1752 = arg2;
-        aCacheArchive_1749 = arg1;
-        aDouble1750 = arg3;
-        anInt1748 = arg4;
+        this.gameImageCacheArchive = gameImageCacheArchive;
+        brightness = arg3;
+        textureSize = arg4;
         anInt1753 = anInt1752;
-        int[] is = arg0.method192(0, true);
-        int i = is.length;
-        aClass40_Sub10Array1740 = new EntryTable[arg0.fileLength(0)];
-        for(int i_4_ = 0; i > i_4_; i_4_++) {
-            Buffer class40_sub1 = new Buffer(arg0.getFile(0, is[i_4_]));
-            aClass40_Sub10Array1740[is[i_4_]] = new EntryTable(class40_sub1);
+        int[] cacheTextures = gameTextureCacheArchive.method192(0, true);
+        int textureCount = cacheTextures.length;
+        textures = new Texture[gameTextureCacheArchive.fileLength(0)];
+        for(int texturePointer = 0; textureCount > texturePointer; texturePointer++) {
+            Buffer textureBuffer = new Buffer(gameTextureCacheArchive.getFile(0, cacheTextures[texturePointer]));
+            textures[cacheTextures[texturePointer]] = new Texture(textureBuffer);
         }
 
     }
 
-    public static Class40_Sub5_Sub15 method421(CacheArchive arg0, byte arg1, int arg2, CacheArchive arg3, boolean arg4) {
+    public static Class40_Sub5_Sub15 method421(CacheArchive skeletonArchive, byte arg1, int arg2, CacheArchive skinArchive, boolean arg4) {
```

Comment:
What is skeletonArchive?

Reply:
animation skeletons

https://www.rune-server.ee/runescape-development/rs2-server/informative-threads/700768-runescapes-animation-system.html